### ISSUE TITLE
Why not `segment.add()`?

### ISSUE THREAD

**Issue by kjohnsen** [NeuralEnsemble/python-neo#1306]
**Is your feature request related to a problem? Please describe.**
I'm trying to take a bunch of data objects and stick them in a segment. Looks like I need to recreate the functionality of `Group.add` which automatically finds the right container to add to if I don't know the class a priori.

**Describe the solution you'd like**
I'd like to be able to call `segment.add(object_of_unknown_class)` like I can with a `Group`. Is there a reason this isn't currently supported? Why isn't `add` defined at the `Container` level instead of `Group`?

**Describe alternatives you've considered**
Copying and pasting `Group`'s code. This is my workaround
```python
def add_to_neo_segment(
    segment: neo.core.Segment, *objects: neo.core.dataobject.DataObject
):
    """Taken from neo.core.group.Group."""
    container_lookup = {
        cls_name: getattr(segment, container_name)
        for cls_name, container_name in zip(
            segment._child_objects, segment._child_containers
        )
    }

    for obj in objects:
        cls = obj.__class__
        if hasattr(cls, "proxy_for"):
            cls = cls.proxy_for
        container = container_lookup[cls.__name__]
        container.append(obj)
```

**Comment by JuliaSprenger** [comment]
Hi @kjohnsen This sound like a very good idea to me. Feel free to open a PR to add the new utility functionality.

**Comment by apdavison** [comment]
Implemented in #1307
